src/features/defi/providers/cosmos/components/CosmosManager/Overview/CosmosOverview.tsx

```diff
@@ -29,14 +31,28 @@ import {
 } from 'state/slices/selectors'
 import { getDefaultValidatorAddressFromAssetId } from 'state/slices/validatorDataSlice/utils'
 import { useAppSelector } from 'state/store'
+import type { Nullable } from 'types/common'
 
 import { CosmosEmpty } from './CosmosEmpty'
 import { WithdrawCard } from './WithdrawCard'
 
-export const CosmosOverview = () => {
+type CosmosOverviewProps = {
+  accountId: Nullable<AccountId>
+  onAccountIdChange: AccountDropdownProps['onChange']
+}
+
+export const CosmosOverview: React.FC<CosmosOverviewProps> = ({
+  accountId: defaultAccountId,
+  onAccountIdChange: handleAccountIdChange,
+}) => {
   const translate = useTranslate()
   const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
-  const { chainId, contractAddress, assetReference } = query
+  const { defaultAccountId: queryAccountId, chainId, contractAddress, assetReference } = query
+
+  const accountId = useMemo(
+    () => defaultAccountId ?? queryAccountId,
+    [defaultAccountId, queryAccountId],
+  )
```

Comment:
This is the guts of d02319b5fe21617053f49ecf3932a12c8a786838 - derp-linking paving the way for deep-linking and removal of `accountSpecifier`.

We will be able to either be fully controlled at manager-level, or get the accountId from queryparams, then be controlled at manager-level